src/LazyDefinition.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Definitions;
+
+use ProxyManager\Factory\LazyLoadingValueHolderFactory;
+use Psr\Container\ContainerInterface;
+use Yiisoft\Definitions\Contract\DefinitionInterface;
+use Yiisoft\Definitions\Helpers\Normalizer;
+
+final class LazyDefinition implements DefinitionInterface
+{
+    public function __construct(
+        private mixed $definition,
+        /**
+         * @var class-string
+         */
+        private string $objectClass,
+    ) {
+    }
+
+    /**
+     * @psalm-suppress MixedArgumentTypeCoercion
+     */
+    public function resolve(ContainerInterface $container): mixed
+    {
+        /** @var LazyLoadingValueHolderFactory $factory */
+        $factory = $container->get(LazyLoadingValueHolderFactory::class);
+        /**
+         * @var mixed $definition
+         */
+        $definition = $this->definition;
+        $objectClass = $this->objectClass;
```

Comment:
Suggest use directly `$this->definition` and `$this->objectClass` without creating new variables.

Reply:
Made the function static